it/uk/gov/hmrc/agentrelationships/connectors/EnrolmentStoreProxyConnectorSpec.scala

```diff
@@ -81,29 +134,83 @@ class EnrolmentStoreProxyConnectorSpec extends UnitSpec with OneServerPerSuite w
 
     "allocate an enrolment to an agent" in {
       givenAuditConnector()
-      givenEnrolmentAllocationSucceeds("group1", "user1", "FOO", "FOO-ID", "ABC1233", "bar")
-      await(connector.allocateEnrolmentToAgent("group1", "user1", Enrolment("FOO", "FOO-ID", "ABC1233"), AgentCode("bar")))
+      givenEnrolmentAllocationSucceeds("group1", "user1", "HMRC-MTD-IT", "MTDITID", "ABC1233", "bar")
+      await(connector.allocateEnrolmentToAgent("group1", "user1", MtdItId("ABC1233"), AgentCode("bar")))
     }
 
-    "throw an exception if allocation failed" in {
+    "throw an exception if allocation failed because of missing agent or enrolment" in {
       givenAuditConnector()
-      givenEnrolmentAllocationFailsWith(404)("group1", "user1", "FOO", "FOO-ID", "ABC1233", "bar")
-      an[Exception] shouldBe thrownBy {
-        await(connector.allocateEnrolmentToAgent("group1", "user1", Enrolment("FOO", "FOO-ID", "ABC1233"), AgentCode("bar")))
+      givenEnrolmentAllocationFailsWith(404)("group1", "user1", "HMRC-MTD-IT", "MTDITID", "ABC1233", "bar")
+      an[NotFoundException] shouldBe thrownBy {
+        await(connector.allocateEnrolmentToAgent("group1", "user1", MtdItId("ABC1233"), AgentCode("bar")))
+      }
+    }
+
+    "throw an exception if allocation failed because of bad request" in {
+      givenAuditConnector()
+      givenEnrolmentAllocationFailsWith(400)("group1", "user1", "HMRC-MTD-IT", "MTDITID", "ABC1233", "bar")
+      an[BadRequestException] shouldBe thrownBy {
+        await(connector.allocateEnrolmentToAgent("group1", "user1", MtdItId("ABC1233"), AgentCode("bar")))
+      }
+    }
+
+    "throw an exception if allocation failed because of unauthorized" in {
+      givenAuditConnector()
+      givenEnrolmentAllocationFailsWith(401)("group1", "user1", "HMRC-MTD-IT", "MTDITID", "ABC1233", "bar")
+      an[Upstream4xxResponse] shouldBe thrownBy {
+        await(connector.allocateEnrolmentToAgent("group1", "user1", MtdItId("ABC1233"), AgentCode("bar")))
+      }
+    }
+
+    "keep calm if conflict reported" in {
+      givenAuditConnector()
+      givenEnrolmentAllocationFailsWith(409)("group1", "user1", "HMRC-MTD-IT", "MTDITID", "ABC1233", "bar")
+      await(connector.allocateEnrolmentToAgent("group1", "user1", MtdItId("ABC1233"), AgentCode("bar")))
```

Comment:
Is there a missing verify or assertion here? Though I guess the point is that it doesn't throw an exception.